lib/find_non_repeated_letter.rb

```diff
@@ -9,5 +9,18 @@
 
 
 def find_first_non_repeated_letter(str)
-  #write your method here
+    ary = str.chars
+    x = 0
+    
+    while x <= ary.length
+        if ary.length <= 1
+            return ary[x]
+        elsif ary[x] == ary[x + 1] || ary[x + 1] == ary[x + 2]
```

Comment:
Is the secondary conditional–`ary[x + 1] == ary[x + 2]` needed? From my perspective it seems like `ary[x] == ary[x + 1]` covers all your bases, but I'm not certain. What am I missing?